post-payments.php

```diff
@@ -0,0 +1,197 @@
+	}
+
+	public function enqueue() {
+		wp_enqueue_script( 'jquery-ui-datepicker' );
+		wp_enqueue_style( 'jquery-ui-style', '//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css' );
+		wp_enqueue_style( 'post-payments-style', plugin_dir_url( __FILE__ ) . 'css/admin.css' );
+
+	}
+
+	public function report_page() {
+		?><div class="wrap">
+			<script type="text/javascript">
+				jQuery(document).ready(function() {
+					jQuery('input.datepicker').datepicker();
+				});
+			</script>
+			<h2><?php esc_html_e( 'Post Payments Report' ); ?></h2>
+			<p><?php esc_html_e( 'Select a date range to view the report.' ); ?></p>
+			<p>
+				<form>
+					<input type="hidden" name="page" value="payments_report" /> 
+					<input name="from_date" class="datepicker" <?php if ( ! empty( $_GET['from_date'] ) ): ?>value="<?php echo esc_attr( $_GET['from_date'] ); ?>"<?php endif; ?> />
+					<input name="to_date" class="datepicker" <?php if ( ! empty( $_GET['to_date'] ) ): ?>value="<?php echo esc_attr( $_GET['to_date'] ); ?>"<?php endif; ?> />
+					<input type="submit" value="<?php esc_html_e( 'Generate' ); ?>" class="button submit" />
+				</form>
+			</p>
+		</div><?php
+
+		if ( ! empty( $_GET['from_date'] ) && ! empty( $_GET['to_date'] ) ) {
+			$authors = $this->get_report_data( sanitize_text_field( $_GET['from_date'] ), sanitize_text_field( $_GET['to_date'] ) );
+
+
+			if ( empty( $authors ) ) {
+				?><p><b><?php esc_html_e( 'No posts found with cost data in this date range.' ); ?></b></p><?php
+			} else {
+				?>
+
+				<?php foreach ( $authors as $name => $data ): ?>
+					<p class="post-payments-author-name"><b><?php echo esc_html( $name ); ?></b></p>
+					<ul class="post-payments-list"><?php foreach ( $data['posts'] as $post_id ): ?>
+						<li><?php echo esc_html( $this->format_currency( get_post_meta( $post_id, $this->meta_key, true ) ) ); ?>: <i><a href="<?php echo esc_url( get_permalink( $post_id ) ); ?>"><?php echo esc_html( get_the_title( $post_id ) ); ?></a></i></li>
+					<?php endforeach; ?></ul>
+					<p class="post-payments-author-payment"><?php esc_html_e( 'Total payment:' ); ?> <b><?php echo esc_html( $this->format_currency( $data['total'] ) ); ?></b></p>
+				<?php endforeach; ?>
+
+				<?php
+			}
+		}
+	}
+
+	public function get_post_types() {
+		$settings = get_option( $this->settings_option );
+		if ( empty( $settings['post_types'] ) ) {
+			return $this->default_post_types;
+		} else {
+			return $settings['post_types'];
+		}
+	}
+
+	public function get_report_data( $from_date, $to_date ) {
+		$posts = get_posts( array(
+			'posts_per_page' => -1,
+			'post_status' => 'publish',
+			'post_type' => $this->get_post_types(),
+			'meta_query' => array(
+				array(
+					'key' => $this->meta_key,
+					'compare' => 'EXISTS',
+				),
+			),
+			'date_query' => array(
+				array(
+					'after' => $from_date,
+					'before' => $to_date,
```

Comment:
I'd considered that, though in reality they will be using it for one month periods on a data pool of a tens of posts. Before it goes public it should have such limits though.